### DISCUSSION TITLE
self-hosted configuration behaviour

### DISCUSSION THREAD

**Discussion by 7opf** [renovatebot/renovate#9102] [Request Help] [answered]
When using self-hosted configuration, do the configurations set there override repository configuration or only act as defaults?

For example if the self-hosted bot is configured with 

```json
  "packageRules": [
    {
      "matchUpdateTypes": ["patch", "pin", "digest"],
      "automerge": true
    }
  ]
```

does this repository configuration get added to the package rules, or override the package rules?
```json
  "packageRules": [
    {
      "matchUpdateTypes": ["major"],
      "automerge": true
    }
  ]
```

**Comment by viceice** [comment, marked as answer]
They got appended, it depends on the config option. some are merged some are simply overwritten.

It's bad practise to configure it in bot config. Better use a shared preset, can be hosted on same platform.

**Reply by 7opf** [reply]
Makes sense. Thanks.
